proxy/src/main/java/com/velocitypowered/proxy/network/ConnectionManager.java

```diff
@@ -1,89 +1,49 @@
 
-    public ConnectionManager(VelocityServer server) {
+    public ConnectionManager(final VelocityServer server) {
         this.server = server;
         this.transportType = TransportType.bestType();
-        this.bossGroup = transportType.createEventLoopGroup(true);
-        this.workerGroup = transportType.createEventLoopGroup(false);
+        this.bossGroup = this.transportType.createEventLoopGroup(TransportType.Type.BOSS);
+        this.workerGroup = this.transportType.createEventLoopGroup(TransportType.Type.WORKER);
         this.logChannelInformation();
     }
 
     private void logChannelInformation() {
-        logger.info("Connections will use {} channels, {} compression, {} ciphers", transportType, Natives.compressor.getLoadedVariant(), Natives.cipher.getLoadedVariant());
+        LOGGER.info("Connections will use {} channels, {} compression, {} ciphers", this.transportType, Natives.compressor.getLoadedVariant(), Natives.cipher.getLoadedVariant());
     }
 
     public void bind(final InetSocketAddress address) {
         final ServerBootstrap bootstrap = new ServerBootstrap()
                 .channel(this.transportType.serverSocketChannelClass)
                 .group(this.bossGroup, this.workerGroup)
                 .childOption(ChannelOption.WRITE_BUFFER_WATER_MARK, SERVER_WRITE_MARK)
-                .childHandler(new ChannelInitializer<Channel>() {
-                    @Override
-                    protected void initChannel(final Channel ch) {
-                        ch.pipeline()
-                                .addLast(READ_TIMEOUT, new ReadTimeoutHandler(server.getConfiguration().getReadTimeout(), TimeUnit.SECONDS))
-                                .addLast(LEGACY_PING_DECODER, new LegacyPingDecoder())
-                                .addLast(FRAME_DECODER, new MinecraftVarintFrameDecoder())
-                                .addLast(LEGACY_PING_ENCODER, LegacyPingEncoder.INSTANCE)
-                                .addLast(FRAME_ENCODER, MinecraftVarintLengthEncoder.INSTANCE)
-                                .addLast(MINECRAFT_DECODER, new MinecraftDecoder(ProtocolConstants.Direction.SERVERBOUND))
-                                .addLast(MINECRAFT_ENCODER, new MinecraftEncoder(ProtocolConstants.Direction.CLIENTBOUND));
-
-                        final MinecraftConnection connection = new MinecraftConnection(ch, server);
-                        connection.setState(StateRegistry.HANDSHAKE);
-                        connection.setSessionHandler(new HandshakeSessionHandler(connection, server));
-                        ch.pipeline().addLast(Connections.HANDLER, connection);
-                        
-                        if (server.getConfiguration().isProxyProtocol()) {
-                            ch.pipeline().addFirst(new HAProxyMessageDecoder());
-                        }
-                    }
-                })
+                .childHandler(new ServerChannelInitializer(this.server))
```

Comment:
Does we really need to create a new object every time?  It's also does not allow to rewrite a ChannelInitializer.